docs/udf-guide.md

```diff
@@ -0,0 +1,159 @@
+# Guide to User-Defined Functions (UDF)
+
+This is a guide to show how to use UDFs in .NET for Apache Spark.
+
+## What are UDFs
+
+[User-Defined Functions (UDFs)](https://spark.apache.org/docs/latest/api/java/org/apache/spark/sql/expressions/UserDefinedFunction.html) are a feature of Spark that allow developers to use custom functions to extend the vocabulary of Spark SQL’s Domain Specific Language. They transform values from a single row within a table to produce a single corresponding output value per row based on the logic defined in the UDF.
```

Comment:
What is Spark SQL's Domain Specific Language? Could we provide a hyperlink and/or explanation to give some more context?

Reply:
Yeah on retrospect I think that is a jargon that doesn't need to be thrown at the user, modifying the sentence to make it more readable.